airflow/models/taskinstance.py

```diff
@@ -1343,18 +1343,27 @@ def _run_finished_callback(self, error: Optional[Union[str, Exception]] = None)
             if task.on_failure_callback is not None:
                 context = self.get_template_context()
                 context["exception"] = error
-                task.on_failure_callback(context)
+                try:
+                    task.on_failure_callback(context)
+                except Exception:
+                    log.exception("Failed when executing execute on_failure_callback:")
         elif self.state == State.SUCCESS:
             task = self.task
             if task.on_success_callback is not None:
                 context = self.get_template_context()
-                task.on_success_callback(context)
+                try:
+                    task.on_success_callback(context)
+                except Exception:
```

Comment:
Nice. I’m suggesting we have a test for this change

Reply:
Great suggestion - I've added a test and in doing so I even found a minor bug which has been fixed now.